x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plan/logical/UnresolvedRelation.java

```diff
@@ -98,16 +95,6 @@ public String unresolvedMessage() {
         return unresolvedMsg;
     }
 
-    @Override
-    public AttributeSet references() {
-        AttributeSet refs = super.references();
-        if (indexMode == IndexMode.TIME_SERIES) {
```

Comment:
Why did you make the move to EsqlSession? Was the `fieldNames` the only place where metrics specific metadata attributes were needed?

Reply:
I think `fieldNames` is the only place where we would somehow call `UnresolvedRelation.references()` and do something with the output in case of time series indices; after the first analysis steps, there's not supposed to be any `UnresolvedRelation`s anymore.

Removing this code altogether makes the tests in `k8s-metrics` fail, whereas it currently runs green; also, the only place in the whole (non-test) code base where `IndexMode.TIME_SERIES` is set on an `UnresolvedRelation` is `LogicalPlanBuilder.visitMetricsCommand`.

I'll double-check if moving this code to `EsqlSession.fieldNames` is the best course of action; if so, I'll add a test case to `IndexResovlerFieldNamesTests`.

Reply:
Ok, it seems like it's reasonable to have a special case for `METRICS` in `EsqlSession.fieldNames`. I added a test.

I also added a comment and made this look for `Aggregate` of `Metrics` type rather than looking for unresolved relations with timeseries mode. It's the `Aggregate` that needs to consume the `@timestamp` field if it's in `Metrics` mode, which I think is easier to grasp than "that's an unresolved time series, so probably we need the `@timestamp` field somewhere".

Reply:
Better to create an issue for this - adding a separate node (or extending Aggregate) is a more elegant way for evolving this requirement.